modules/post/windows/gather/enum_chocolatey_applications.rb

```diff
@@ -0,0 +1,69 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+                      "Name" => "Windows Gather Installed Application Within Chocolatey Enumeration",
+                      "Description" => %q{ This module will enumerate all installed applications on a Windows system with chocolatey installed },
+                      "License" => MSF_LICENSE,
+                      "Author" => ["Nick Cottrell <ncottrellweb@gmail.com>"],
+                      "Platform" => ["win"],
+                      "SessionTypes" => ["meterpreter", "shell"]))
+  end
+
+  def have_chocolatey?
+    cmd_exec("choco.exe", "-v") =~ /\d+\.\d+\.\d+/m
```

Comment:
I think this makes some presumptions about the `%PATH%` variable of an attackers session being set.
Might be handy to make it a DS option like we do with powershell and a bunch of other stuff to permit full pathing, including on OS which are not running Latin character sets.vs

Reply:
I just added an advanced option to specify the location of the binary (with default just using path). If you know where I can find the method that you're referring to, I'd happily reorganize my code to fit it